posthog/hogql_queries/web_analytics/stats_table.py

```diff
@@ -309,6 +312,66 @@ def to_path_bounce_query(self) -> ast.SelectQuery:
 
         return query
 
+    def to_frustration_metrics_query(self) -> ast.SelectQuery:
+        with self.timings.measure("frustration_metrics_query"):
+            # Base selects, always returns the breakdown value, and the total number of visitors
+            selects = [
+                ast.Alias(alias="context.columns.breakdown_value", expr=self._processed_breakdown_value()),
+                self._period_comparison_tuple("rage_clicks_count", "context.columns.rage_clicks", "sum"),
+                self._period_comparison_tuple("dead_clicks_count", "context.columns.dead_clicks", "sum"),
+                self._period_comparison_tuple("errors_count", "context.columns.errors", "sum"),
+            ]
+
+            query = ast.SelectQuery(
+                select=selects,
+                select_from=ast.JoinExpr(table=self._frustration_metrics_inner_query()),
+                group_by=[ast.Field(chain=["context.columns.breakdown_value"])],
+                order_by=self._frustration_metrics_order_by(columns=[select.alias for select in selects]),
+            )
+
+        return query
+
+    def _frustration_metrics_inner_query(self):
+        query = parse_select(
```

Comment:
nit: This query is simple enough for you to build using `ast.SelectQuery` for some extra performance

Reply:
That's true, the first attempt was just a tile with the hogql query and it was way simpler, but it required more changes to make the cross_sell stuff work: https://github.com/PostHog/posthog/pull/31102/commits/e766a8fe172a1ac3d4934f48dd615c92d5fe6a31 so this is here just to keep the pattern as the other web_stats_table queries.